resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentServiceImpl.java

```diff
@@ -1616,8 +1608,7 @@ public byte[] getUINCard(String rid) {
 							ResidentErrorCode.DIGITAL_CARD_RID_NOT_FOUND.getErrorMessage());
 				}
 				URI dataShareUri = URI.create(digitalCardStatusResponseDto.getUrl());
-				String encryptedData = residentServiceRestClient.getApi(dataShareUri, String.class);
-				return decryptDataShareData(encryptedData);
+				return residentServiceRestClient.getApi(dataShareUri, byte[].class);
```

Comment:
Have a flag to use the decryption logic. Disable it by default.

Reply:
Done.